core/src/main/java/io/questdb/cairo/wal/seq/TableSequencerAPI.java

```diff
@@ -319,6 +326,40 @@ private TableSequencerEntry openSequencerLocked(CharSequence tableName, Sequence
         return getTableSequencerEntry(Chars.toString(tableName), lock, this.openSequencerInstanceLambda);
     }
 
+    private void readWalTableInfo(
+            CharSequence tableName,
+            WalTableInfo tableInfo,
+            LastWalTxnReader lastTxnReader,
+            Path path,
+            int rootLen
+    ) {
+        tableInfo.reset();
+        // Fast path.
+        // We don't need to allocate a String for the table name here since we don't create a new map entry.
+        seqRegistry.compute(tableName, (name, sequencer) -> {
+            if (sequencer != null) {
+                if (!sequencer.isDistressed() && !sequencer.isClosed()) {
+                    tableInfo.tableId = sequencer.getTableId();
+                    tableInfo.lastTxn = sequencer.lastTxn();
+                }
+                return sequencer;
+            }
+            try (LastWalTxnReader ignore = lastTxnReader) {
+                lastTxnReader.open(path, rootLen);
+                tableInfo.tableId = lastTxnReader.getTableId();
+                tableInfo.lastTxn = lastTxnReader.lastTxn();
+            }
+            return null;
+        });
+        // Slow path.
+        if (tableInfo.tableId == -1) {
+            try (TableSequencer tableSequencer = openSequencerLocked(tableName, SequencerLockType.NONE)) {
```

Comment:
Why can we read these 2 values from files instead of creating the TableSequencer instance?

Reply:
Slow path doesn't create a `TableSequencer` instance. The condition `tableInfo.tableId == -1` would hold only if we there is a distressed or closed instance already. See the above `compute()` call.